test/test-parse-statement.cpp

```diff
@@ -1244,6 +1244,61 @@ TEST_F(Test_Parse_Statement, disallow_label_named_yield_in_generator_function) {
                             "visit_exit_function_scope",
                         }));
 }
+
+TEST_F(Test_Parse_Statement, if_body_with_semicolon_typescript) {
+  {
+    Spy_Visitor p = test_parse_and_visit_statement(
+        u8"if (a)\n; else e;"_sv,  //
+        u8"      ` Diag_Missing_Body_For_If_Statement"_diag,
```

Comment:
Minor: I think the span is a bit confusing. If the user writes a body at the suggested point, they will end up with code like this:
```javascript
if (a) {
}
; else e;
```

I think we should instead highlight the `;` and hint that the `;` needs to be replaced with a body:
```javascript
if (a) {
} else e;
```

```suggestion
        u8"if (a)\n; else e;"_sv,  //
        u8"        ^ Diag_Missing_Body_For_If_Statement"_diag,
```